billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AbstractStoragePricingService.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.billingreportagent.service.impl.converter;
+
+import com.epam.pipeline.billingreportagent.model.billing.StoragePricing;
+import com.epam.pipeline.entity.region.CloudProvider;
+import lombok.extern.slf4j.Slf4j;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@SuppressWarnings("checkstyle:magicNumber")
+public abstract class AbstractStoragePricingService {
```

Comment:
Currently this abstract class contains lot's of implemented methods and two abstract methods. Can we make `AbstractStoragePricingService` just `StoragePricingService` and extract all abstract methods into a separate interface with implementations for different providers? I think it will make code more readable.

Reply:
Done, introduced `StoragePriceListLoader` interface